download-utilities.sh

```diff
@@ -483,6 +483,12 @@ process_args() {
         if [ -z "${yaml_file:-}" ]; then
           yaml_file="$1"
           shift
+          
+          if [[ "${yaml_file}" == "-" ]]; then
+            temp_file=$(mktemp)
+            cat > "${temp_file}"
+            yaml_file=${temp_file}
+          fi
```

Comment:
Move this code to line 533 and use the provided $TMP_DIR environment variable.  That directory will automatically delete when the script completes.